llvm/lib/Target/AMDGPU/SIMachineFunctionInfo.cpp

```diff
@@ -107,8 +107,8 @@ SIMachineFunctionInfo::SIMachineFunctionInfo(const Function &F,
       MayNeedAGPRs = false; // We will select all MAI with VGPR operands.
   }
 
-  if (!AMDGPU::isGraphics(CC) ||
-      (CC == CallingConv::AMDGPU_CS && ST.hasArchitectedSGPRs())) {
+  if (!AMDGPU::isGraphics(CC) || CC == CallingConv::AMDGPU_CS ||
+      ST.hasArchitectedSGPRs()) {
```

Comment:
This change looks redundant as this patch always allocate the TTMP* for subtargets with architectedSGPR enabled.
This was initially added with https://github.com/llvm/llvm-project/commit/2171f04c121ed27fe2705d3f116df052daeb40a5.
You can revert this check to just have only !AMDGPU::isGraphics(CC)

Reply:
But I need `WorkGroupIDZ` to be set correctly when architected SGPRs are enabled. It is used below, line 177.